v2/tools/generator/internal/codegen/pipeline/apply_export_filters.go

```diff
@@ -32,47 +31,69 @@ func filterTypes(
 	configuration *config.Configuration,
 	state *State) (*State, error) {
 
-	newDefinitions := make(astmodel.Types)
-
-	filterer := configuration.BuildExportFilterer(state.types)
+	defs := make(astmodel.Types)
+	types := state.Types()
 	renames := make(map[astmodel.TypeName]astmodel.TypeName)
-
-	for _, def := range state.types {
+	for _, def := range astmodel.FindResourceTypes(types) {
 		defName := def.Name()
-		shouldExport, newName, reason := filterer(defName)
-
-		switch shouldExport {
-		case config.Skip:
-			klog.V(3).Infof("Skipping %s because %s", defName, reason)
 
-		case config.Export:
-			if reason == "" {
-				klog.V(3).Infof("Exporting %s", defName)
-			} else {
-				klog.V(2).Infof("Exporting %s because %s", defName, reason)
+		export, err := configuration.ObjectModelConfiguration.LookupExport(defName)
+		if err != nil {
+			if config.IsNotConfiguredError(err) {
+				export = false
 			}
+		}
+
+		if _, err = configuration.ObjectModelConfiguration.LookupExportAs(defName); err == nil {
```

Comment:
minor: Wondering if it would be slightly clearer if this second bit was also done by `LookupExport` (possibly with a changed name), so that the contract w/ the configuration was just `config.IsThisThingSupposedToBeExported()`, as opposed to having to stitch together a couple properties that actually each mean that in their own way

Reply:
I thought about that, but decided the `config` package should be about configuration - not so much about the business rules. I'll move the logic into new function, `shouldExport()` which should make things clearer.